### PR TITLE
Add persistence in generated resources

### PR DESCRIPTION
**Pull request by gabriel-farache**
https://issues.redhat.com/browse/FLPATH-1177

- Add persistence extensions when generating the workflows images
- Add persistence spec in manifests
  - the spec is removed for e2e tests

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ jobs:
           pipeline/${{ env.DOCKERFILE }}
         build-args: |
           WF_RESOURCES=${{ inputs.workflow_id }}/
-          QUARKUS_EXTENSIONS=org.kie.kogito:kogito-addons-quarkus-jobs-knative-eventing:999-SNAPSHOT
+          QUARKUS_EXTENSIONS=org.kie.kogito:kogito-addons-quarkus-jobs-knative-eventing:999-SNAPSHOT,org.kie.kogito:kogito-addons-persistence-jdbc:9.99.0.redhat-00007,io.quarkus:quarkus-jdbc-postgresql:3.2.9.Final,io.quarkus:quarkus-agroal:3.2.9.Final
```
Comment (masayag) [inline .github/workflows/main.yml:43]: is there a need to enable the `earlyaccess` repo?

https://maven.repository.redhat.com/earlyaccess/all/org/kie/kogito/kogito-addons-quarkus-persistence-jdbc/9.99.0.redhat-00007/

Comment (gabriel-farache) [reply .github/workflows/main.yml:43]: yes, I was about to ask about that

Comment (gabriel-farache) [reply .github/workflows/main.yml:43]: @masayag I added an ugly hack in the dockerfiles and now it should work (locally with make it works)

**Review by dmartinol — APPROVED** [review]

File: pipeline/workflow-builder.Dockerfile [hunk 2]
```diff
@@ -16,6 +16,9 @@ RUN ls -la ./resources
 
 ENV swf_home_dir=/home/kogito/serverless-workflow-project
 RUN if [[ -d "./resources/src" ]]; then cp -r ./resources/src/* ./src/; fi
+# Hacking pom.xml to add early access repo
```
Comment (masayag) [inline pipeline/workflow-builder.Dockerfile:19]: pls replace this hack with:
```
# Temp hack to provide persistence artifacts
ENV MAVEN_REPO_URL=https://maven.repository.redhat.com/earlyaccess/all
```

Comment (gabriel-farache) [reply pipeline/workflow-builder.Dockerfile]: Done

https://github.com/parodos-dev/serverless-workflows/issues/167 to remove the fix

**Review by masayag — APPROVED** [review]

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -86,6 +86,11 @@ jobs:
       run: yq -i 'del(.metadata.annotations."sonataflow.org/profile")' ${{ inputs.workflow_id }}/manifests/01-sonataflow*.yaml
     - name: Set container image ref in SonataFlow resource
       run: yq -i '.spec.podTemplate.container.image="quay.io/orchestrator/serverless-workflow-${{ inputs.workflow_id }}:${{ github.sha }}"' ${{ inputs.workflow_id }}/manifests/01-sonataflow*.yaml
+    - name: Set persistence spec in SonataFlow resource
```
Comment (dmartinol) [inline .github/workflows/main.yml:89]: just asking: 
1. do we need a similar patch in serverless-workflows-config for kustomize deployments?
1. and in the Tekton pipeline triggered by software workflows?

**Review by dmartinol — APPROVED** [review]

File: .github/workflows/main.yml (hunk 3 above)
Comment (gabriel-farache) [reply .github/workflows/main.yml:89]: > and in the Tekton pipeline triggered by software workflows?

https://github.com/parodos-dev/orchestrator-helm-chart/pull/127

> do we need a similar patch in serverless-workflows-config for kustomize deployments?

Yes, good point

**Pull request merged**